core/server/master/src/main/java/alluxio/master/AlluxioMasterProcess.java

```diff
@@ -220,20 +220,16 @@ protected void stopMasters() {
    * server and starting web ui.
    */
   protected void startServingWebServer() {
-    try {
-      mWebServer =
-          new MasterWebServer(ServiceType.MASTER_WEB.getServiceName(),
-              getWebAddressFromBindSocket(), this);
-      // reset master web port
-      // Add the metrics servlet to the web server.
-      mWebServer.addHandler(mMetricsServlet.getHandler());
-      // Add the prometheus metrics servlet to the web server.
-      mWebServer.addHandler(mPMetricsServlet.getHandler());
-      // start web ui
-      mWebServer.start();
-    } catch (IOException e) {
-      throw new RuntimeException(e);
-    }
+    stopRejectingWebServer();
```

Comment:
If the rejecting web server fails to stop, wouldn't it be more ideal to catch the exception and then re-throw a RuntimeException or something similar here instead of just waiting for the MasterWebServer to fail on a port conflict? I could see the error being confusing if the web server fails to start because of a port conflict even if nothing is running on the port once the process stops.

Reply:
Ah after tracing again I can see now